src/isl/isl_type.rs

```diff
@@ -109,6 +199,13 @@ impl TryFrom<&OwnedElement> for IslType {
             };
             constraints.push(constraint);
         }
-        Ok(IslType::new(type_name, constraints))
+        Ok(NamedIslType::new(type_name, constraints))
+    }
+}
+
+/// Provides an implementation of PartialEq to compare two IslTypes
+impl PartialEq for NamedIslType {
+    fn eq(&self, other: &Self) -> bool {
+        self.name == other.name && self.constraints.eq(&other.constraints)
```

Comment:
This looks like it's the same `PartialEq` implementation that we'd get from `#[derive(PartialEq)]`. If that's true, consider deriving it instead. Same thought for `AnonymousIslType`'s `PartialEq` impl.